### ISSUE TITLE
INSTALL doc instructions are not updated for Building Correctly Libmtp manually

### ISSUE THREAD

**Issue by douggynix** [libmtp/libmtp#142]
I wanted to build manually Libmtp by following the instructions from the INSTALL DOC.
If a user follows exactly what's in there. it won't work as i notice that you're using autoconf to generate the configure script. And it seems like this is not mentionned in the documentation. Not everybody is familiar with Autotools. 
it would be more convenient if the instructions are provided for the one who'd like to build the program themselves.
Here is how I proceed by running these commands below in the exact order :
```
aclocal
autoconf
autoreconf -vsi --force
automake --add-missing
```

Then i was able to see the `configure` script created. Going for the next step to run configure, it results in an error:
`./configure `
configure: error: cannot find required auxiliary files: config.rpath ltmain.sh

it complains about an expected file which is missing. To bypass this issue , I just had to create the two files above with  touch command **config.rpath**  and ltmain.sh :
`touch ./config.rpath ltmain.sh ./ltmain.sh`

And then rerun 
```
./configure
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a race-free mkdir -p... /usr/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes

```
Then it works. Would you please update the INSTALL doc with the detailed instructions of how to build it. I reproduce the same problem many times. You can also use my hints above if needed.
All of the above have been reproduced with the latest version of the code we have on master branch.
I am using Archlinux.

And finally to build it, i run make :
```
make
make install 
```

In summary, for a successful reproducible build of libmtp by any individual, those instructions below have it working successfully. The install doc should be adjusted accordingly as below:
```
touch ./config.rpath ltmain.sh ./ltmain.sh
aclocal
autoconf
autoreconf -vsi --force
automake --add-missing
./configure
make
make install 
```

**Comment by msmeissn** [comment]
i think this might happen only if libtool is not correctly installed

**Comment by JoesCat** [comment]
@douggynix - check to see if libtool is installed, it appears to reduce down to:
```c
touch ./config.rpath
aclocal
autoconf
autoreconf -i
automake
./configure
make
make install
```
Searching for info about config.rpath, I found this:
[[sr #110435] autoreconf does not recognize AM_ICONV](https://lists.gnu.org/archive/html/bug-autoconf/2021-01/msg00033.html) which may explain the config.rpath.
Also noticed file "m4/iconv.m4" has a line "AC_REQUIRE([AC_LIB_RPATH])"

Looking at autogen.sh, it already appears it contains a line to touch config.rpath

**Comment by douggynix** [comment]
@JoesCat @msmeissn
Thanks for your feedback and comments about libtool missing. 
The main point about the issue is not about Libtool itself. it's the fact that The INSTALL doc doesn't mention that you have to use Autoconf and automake to build Libmtp. And the instructions are outdated. 
In the install doc, it is mentioned that you should run a ./configure script which doesn't exist. and for it to exist, you have to run in advance all these instructions above.

The Install doc must be updated to be in sync with the build tools used by Libmtp project. That's what needed to be changed.
https://github.com/libmtp/libmtp/blob/master/INSTALL
![image](https://user-images.githubusercontent.com/11457752/221463448-4d843501-b819-467e-8da7-a1c7a4ab29d9.png)

**Comment by cgimenez** [comment]
I was against the same wall on macOS 10.15 with MacPorts - Thanks for this issue I could build libmtp by following the above instructions with one single change :
`./configure LDFLAGS="-L/opt/local/lib"`

**Comment by msmeissn** [comment]
the description was there, just not in the right timewise order. I reordered it a bit.
